### PR TITLE
Make Telemetry an interface

### PR DESCRIPTION
**Pull request by remcohaszing**
### What does this PR do?

In `monaco-yaml` I’m interested in stubbing the Telemetry option. However, currently everywhere this is used uses a type annotation based on the class. This means the implementation must match this class, including private properties. By accepting an interface instead, it’s possible to pass alternative implementations.

### What issues does this PR fix or reference?

https://github.com/remcohaszing/monaco-yaml/issues/190

### Is it tested? How?

By running `tsc`.

### CODE REVIEW

**Comment by msivasubramaniaan** [comment]
@remcohaszing Update the branch

**Comment by remcohaszing** [comment]
Done :+1:

**Review by msivasubramaniaan — DISMISSED** [review]
LGTM

File: src/languageserver/telemetry.ts [hunk 1]
```diff
@@ -4,20 +4,9 @@
...
-  name: string;
-  type?: string;
-  properties?: unknown;
-  measures?: unknown;
-  traits?: unknown;
-  context?: unknown;
-}
-
-export class Telemetry {
+export class Telemetry implements TelemetryType {
```
[the remark this answers is not in the document]
Comment (remcohaszing) [reply src/languageserver/telemetry.ts]: Sure! That makes sense.

**Review by fbricon — APPROVED** [review]

**Pull request merged**